proxy_agent/src/common/error.rs

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+
+use std::error::Error as StdError;
+use std::fmt::Display;
+
```

Comment:
nit: remove extra line break. it would be nice keep all the use statements together.